dags/springer/parser.py

```diff
@@ -118,13 +129,15 @@ def _get_abstract(self, article: ET.Element):
             [child_text for child_text in child.itertext()] for child in paragraph
         ]
         text_to_skip_flatten = [
-            item for sublist in text_to_skip_arrays for item in sublist
+            item
+            for sublist in text_to_skip_arrays
+            for item in sublist
+            if not is_latex(item)
```

Comment:
I just realize this code, we can make it more readable with recursion for an example can be:

```python
def flatten_list(input_list):
      return [
          element for innerList in input_list for element in flatten_list(innerList)
      ]
```

Reply:
I don't totally get the suggestion. Recursion needs a ending condition. Here there is none and we will be trying to iterate through a int and it will fail. Am I missing something ?

Reply:
my bad I didn't completely finished my suggestion:

```python
def flatten_list(input_list):
    if isinstance(input_list, (list, tuple)):
        return [
            element for innerList in input_list for element in flatten_list(innerList)
        ]
    return [input_list]
```

Reply:
I find that your code is the same readability than mine (because we still have 2 `for`). Yours is good to handle lists that have more than two levels which is not the case here. Do you think that I should still change it ?

Reply:
I was having having trouble to understand why we need too many loops (including `text_to_skip_arrays`), maybe i'm missing something, but can we just reduce them by doing the following:

```python
# no need of `text_to_skip_arrays` and 
text_to_skip_flatten = [
     child_text 
     for child in paragraph 
     for child_text in child.itertext() 
     if not is_latex(child_text)
]
```

One more question instead of `child.itertext` can we just get `chid.text` why we need to iterate? this will reduce the loops to 1:

```python
text_to_skip_flatten = [
     text 
     for text in paragraph
     if not is_latex(child_text)
]

# or 
filter(lambda x: not is_latex(x), paragraph)
```

WDYT?

Reply:
Reasoning for `text_to_skip_arrays`: All children do not have valuable data. We need to first identify the data that we don't need to skip it.

Reasoning for `child.itertext`: `child.text` only returns the text from the current node. `itertext` returns from current node and from children, and children inside it.

I will check again this function anyways to see if it cannot be rewritten in some way

Reply:
Sorry I didn't get your point on text to skip before. Changed, let me know if it's better 👍